app/services/katello/repository_type_manager.rb

```diff
@@ -140,6 +140,14 @@ def enabled?(repository_type)
         find(repository_type).present?
       end
 
+      def generic_remote_options
+        options = []
+        enabled_repository_types.each do |_, type|
```

Comment:
I'm working on an issue right now where params in controllers shouldn't be based on enabled repo types. It's because, at rpm build time, when the api cache is generated, the enabled types will be empty. I would give an argument for enabled_only (`def generic_remote_options(enabled_only = true)`.  Then, if `enabled_only` is false, use `defined_repository_types` instead of `enabled_repository_types`.  That way, you can still use generic_remote_options for cases where you want to use the enabled types (which should be pretty much any case other than when you're creating the controller params).

Reply:
Example: https://github.com/Katello/katello/pull/9480/files#diff-9d7db34467c739b369c12749ec9e0f66c590619691fb5921a6f6ef42f201a8bdR48